vmm/sandbox/src/stratovirt/mod.rs

```diff
@@ -450,11 +450,13 @@ impl StratoVirtVM {
         &mut self,
         mut device: T,
     ) -> Result<()> {
-        // get the empty slot index
-        let slot_index = self.get_empty_pcie_slot_index()?;
-        // set the pcie slot status to Occupied
-        self.pcie_root_bus.bus.slots[slot_index].status = SlotStatus::Occupied(device.id());
-        device.set_device_addr(slot_index);
+        if !self.config.machine.r#type.contains(MACHINE_TYPE_MICROVM) {
```

Comment:
this makes the machine type of `microvm` a special case, can we just make the `pcie_root_bus` of `StratoVirtVM` an Option so that we just need to  check if it is a None here.

Reply:
fixed